aws-api-appsync/src/test/resources/serialized-model-with-metadata.json

```diff
@@ -10,7 +10,8 @@
     "createdAt":"2022-02-19T00:05:33.564Z",
     "id":"21ee0180-60a4-45d9-b68e-018c260cc742",
     "_version":3,
-    "updatedAt":"2022-03-04T05:36:26.629Z"
+    "updatedAt":"2022-03-04T05:36:26.629Z",
+    "__typename": "Post"
```

Comment:
what makes adding `__typename` now a required change for CPK? based on that, i wonder if we could make an incremental change to the codebase to have a PR that only adds `__typename` to set us up for a smaller CPK PR

Reply:
We need this because earlier for deserializing the flutter model we assumed primary key field to be "hardcoded", to support the cpk we need to know the field names and hence require the type name to get the schema.